internal/cmd/root/root.go

```diff
@@ -14,12 +17,36 @@ import (
 	"github.com/spf13/cobra"
 )
 
+type options struct {
+	Verbose bool
+}
+
+func (o *options) AddFlags(cmd *cobra.Command) {
+	cmd.PersistentFlags().BoolVar(&o.Verbose, "verbose", false, "Enable verbose logging")
+}
+
 func New() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "gittuf",
 		Short: "A security layer for Git repositories, powered by TUF",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			level := slog.LevelInfo
+			isVerbose, err := cmd.Flags().GetBool("verbose")
```

Comment:
if `o := &options{}` is moved up above the `cmd` definition, we can just use `o.verbose` right?

Reply:
We shouldn't have an error then and we can just use `PersistentPreRun`?

Reply:
```
	o := &options{}
	cmd := &cobra.Command{
		Use:   "gittuf",
		Short: "A security layer for Git repositories, powered by TUF",
		PersistentPreRun: func(cmd *cobra.Command, args []string) {
			level := slog.LevelInfo
			if o.verbose {
				level = slog.LevelDebug
			}

			slog.SetDefault(slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{Level: level})))
		},
	}
	o.AddFlags(cmd)
```

Reply:
Yeah. Updated and rebased.